pkg/operator/ceph/file/controller.go

```diff
@@ -254,12 +286,62 @@ func (r *ReconcileCephFilesystem) reconcile(request reconcile.Request) (reconcil
 	logger.Debug("reconciling ceph filesystem store deployments")
 	reconcileResponse, err = r.reconcileCreateFilesystem(cephFilesystem)
 	if err != nil {
-		updateStatus(r.client, request.NamespacedName, k8sutil.ReconcileFailedStatus)
+		updateStatus(r.client, request.NamespacedName, cephv1.ConditionFailure, nil)
 		return reconcileResponse, err
 	}
 
-	// Set Ready status, we are done reconciling
-	updateStatus(r.client, request.NamespacedName, k8sutil.ReadyStatus)
+	// Enable mirroring if needed
+	// TODO: change me to 16.2.5 once it's out, in the mean this allows us to run the CI and validate the code
+	if r.clusterInfo.CephVersion.IsAtLeastPacific() {
+		// Disable mirroring on that filesystem if needed
+		if cephFilesystem.Spec.Mirroring != nil {
+			if !cephFilesystem.Spec.Mirroring.Enabled {
+				err = cephclient.DisableFilesystemSnapshotMirror(r.context, r.clusterInfo, cephFilesystem.Name)
+				if err != nil {
+					return reconcile.Result{}, errors.Wrapf(err, "failed to disable mirroring on filesystem %q", cephFilesystem.Name)
+				}
+			} else {
+				if cephFilesystem.Spec.Mirroring.IsEnabled() {
```

Comment:
Could `IsEnabled()` be moved up to line 297? It seems like these checks whether mirroring is enabled could be simplified that way. Or maybe just this line seems unnecessary since we're in the `else` statement that should already mean mirroring is enabled.

Reply:
Yeah and also `IsEnabled()` is kinda useless too.